src/algebraic_geometry/projective_spectrum/topology.lean

```diff
@@ -124,4 +124,321 @@ lemma subset_zero_locus_iff_le_vanishing_ideal (t : set (projective_spectrum 
+  x ∈ (↑(basic_open 𝒜 f): set (projective_spectrum 𝒜)) ↔ f ∉ x.as_homogeneous_ideal := iff.rfl
+
+lemma is_open_basic_open {a : A} : is_open ((basic_open 𝒜 a) :
+  set (projective_spectrum 𝒜)) :=
+(basic_open 𝒜 a).property
+
+@[simp] lemma basic_open_eq_zero_locus_compl (r : A) :
+  (basic_open 𝒜 r : set (projective_spectrum 𝒜)) = (zero_locus 𝒜 {r})ᶜ :=
+set.ext $ λ x, by simpa only [set.mem_compl_eq, mem_zero_locus, set.singleton_subset_iff]
+
+@[simp] lemma basic_open_one : basic_open 𝒜 (1 : A) = ⊤ :=
+topological_space.opens.ext $ by simp
+
+@[simp] lemma basic_open_zero : basic_open 𝒜 (0 : A) = ⊥ :=
+topological_space.opens.ext $ by simp
+
+lemma basic_open_mul (f g : A) : basic_open 𝒜 (f * g) = basic_open 𝒜 f ⊓ basic_open 𝒜 g :=
+topological_space.opens.ext $ by {simp [zero_locus_singleton_mul]}
+
+lemma basic_open_mul_le_left (f g : A) : basic_open 𝒜 (f * g) ≤ basic_open 𝒜 f :=
+by { rw basic_open_mul 𝒜 f g, exact inf_le_left }
+
+lemma basic_open_mul_le_right (f g : A) : basic_open 𝒜 (f * g) ≤ basic_open 𝒜 g :=
+by { rw basic_open_mul 𝒜 f g, exact inf_le_right }
+
+@[simp] lemma basic_open_pow (f : A) (n : ℕ) (hn : 0 < n) :
+  basic_open 𝒜 (f ^ n) = basic_open 𝒜 f :=
+topological_space.opens.ext $ by simpa using zero_locus_singleton_pow 𝒜 f n hn
+
+lemma basic_open_eq_union_of_projection (f : A) :
+  basic_open 𝒜 f = ⨆ (i : ℕ), basic_open 𝒜 (graded_algebra.proj 𝒜 i f) :=
+topological_space.opens.ext $ set.ext $ λ z, begin
+  erw [mem_coe_basic_open, topological_space.opens.mem_Sup],
+  split; intros hz,
+  { rcases show ∃ i, graded_algebra.proj 𝒜 i f ∉ z.as_homogeneous_ideal, begin
+      contrapose! hz with H,
+      haveI : Π (i : ℕ) (x : 𝒜 i), decidable (x ≠ 0) := λ _, classical.dec_pred _,
+      rw ←graded_algebra.sum_support_decompose 𝒜 f,
+      apply ideal.sum_mem _ (λ i hi, H i)
+    end with ⟨i, hi⟩,
+    exact ⟨basic_open 𝒜 (graded_algebra.proj 𝒜 i f), ⟨i, rfl⟩, by rwa mem_basic_open⟩ },
+  { obtain ⟨_, ⟨i, rfl⟩, hz⟩ := hz,
+    exact λ rid, hz (z.1.2 i rid) },
+end
+
+lemma is_topological_basis_basic_opens : topological_space.is_topological_basis
+  (set.range (λ (r : A), (basic_open 𝒜 r : set (projective_spectrum 𝒜)))) :=
+begin
+  apply topological_space.is_topological_basis_of_open_of_nhds,
+  { rintros _ ⟨r, rfl⟩,
+    exact is_open_basic_open 𝒜 },
+  { rintros p U hp ⟨s, hs⟩,
+    rw [← compl_compl U, set.mem_compl_eq, ← hs, mem_zero_locus, set.not_subset] at hp,
+    obtain ⟨f, hfs, hfp⟩ := hp,
+    refine ⟨basic_open 𝒜 f, ⟨f, rfl⟩, hfp, _⟩,
+    rw [← set.compl_subset_compl, ← hs, basic_open_eq_zero_locus_compl, compl_compl],
+    exact zero_locus_anti_mono 𝒜 (set.singleton_subset_iff.mpr hfs) }
+end
+
+end basic_open
+
+section order
+
+/-!
+## The specialization order
+
+We endow `projective_spectrum 𝒜` with a partial order,
+where `x ≤ y` if and only if `y ∈ closure {x}`.
+
+TODO: maybe define sober topological spaces, and generalise this instance to those
```

Comment:
Sober spaces have been defined. The `prime_spectrum` already has an instance.
```
src/topology/sober.lean
12:A quasi-sober space is a topological space where every
14:A sober space is a quasi-sober space where every irreducible closed subset
15:has a *unique* generic point. This is if and only if the space is T0, and thus sober spaces can be
16:stated via `[quasi_sober α] [t0_space α]`.
```

Reply:
I think it might make more sense if I do this in a separate pr.